### DISCUSSION TITLE
Feature Request

### DISCUSSION THREAD

**Discussion by TonyginSLC** [aclap-dev/video-downloadhelper#1321] [Help and Q&A]
### Your question

I would love to see the next two features in an upcoming release: 1) the abilities to set retries on download failures, and 2) when it download it does fail, the ability to see on which exact file it failed.
![Screenshot_20241002_090204](https://github.com/user-attachments/assets/b87386c7-0a5d-42e9-9097-db1282fbf90e)

### Addon & CoApp version and details

The Firefox add in is version 9.1.0.41. I don't know how to determine it the CoApp version but it is probably the most recent one.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Good ideas.

For your suggestion `#2` I think the new history feature might be of some help.  I've made a suggestion similar to this somewhere on here.  But I've suggested that instead of just saying click this & some unknown data is sent to some unknown destination, we should be able to see whatever it is that is being sent.  There is a small probability that we dumb users might occasionally be able to interpret whatever is hiding behind that `Report` button & solve our own problems.  Occasionally.  Maybe.  I don't like the idea that data I have no access to is sent to a destination I can't identify.  The destination doesn't bother me too much because I do trust VDH.  But not being able to even see the data is a horrible design.  We should be able to see what is sent.  When you send a crashdump to Mozilla, you can look at it, for what that's worth.  But at least you can read it.  We aren't given that option with this error reporting mechanism.  That stinks.  That sucks.  That sort of bad design is the result of the complete absence of walkthroughs in the VDH development process.  A second opinion is always desirable when you're developing software.  Usually, there's also a third & fourth opinion.  Oh but that slows down the development process.  So?  This is bad how?  Better to have better software after an insignificant delay than bad software.  This reporting mechanism is bad software.  I would have objected to it had I been given the chance before it was implemented.  But I'm not in that loop.

And here's something you might find useful:

![#01](https://github.com/user-attachments/assets/ddd313ba-e372-497a-8f47-965ba91ac3ab)
![#02](https://github.com/user-attachments/assets/c94603f4-3eb5-4001-9aca-262d24811ea0)
![#03](https://github.com/user-attachments/assets/2bc1be17-94e4-43fa-ae76-e0149cf4258d)
